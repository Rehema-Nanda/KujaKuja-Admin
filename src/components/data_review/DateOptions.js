export default class DateOptions {
    static get CREATED_AT() {
        return "Created At";
    }

    static get UPLOADED_AT() {
        return "Uploaded At";
    }

    static get ALL_OPTIONS() {
        return [this.CREATED_AT, this.UPLOADED_AT];
    }
}
