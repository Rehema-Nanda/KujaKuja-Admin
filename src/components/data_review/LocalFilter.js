import React, { Component } from "react";
import {
    Navbar, NavbarBrand, NavbarToggler, Collapse, Button, Input, Label, FormGroup,
} from "reactstrap";
import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";

import LocalDateFilter from "./LocalDateFilter";
import LocalUserLocationFilter from "./LocalUserLocationFilter";
import DateOptions from "./DateOptions";

class LocalFilter extends Component {
    constructor(props) {
        super(props);

        this.state = {
            showMobileFilter: true,
            mobileControlsOpen: false,
            showMobileFilterDates: false,
            showMobileFilterLocations: false,
        };
    }

    toggleMobileControlsAndClose = () => {
        const { mobileControlsOpen } = this.state;

        this.setState({
            mobileControlsOpen: !mobileControlsOpen,
        });
    };

    displayMobileFilter = () => {
        this.setState(
            {
                showMobileFilter: true,
                showMobileFilterLocations: false,
                showMobileFilterDates: false,
            },
        );
    };

    displayMobileFilterLocations = () => {
        this.setState(
            {
                showMobileFilter: false,
                showMobileFilterLocations: true,
                showMobileFilterDates: false,
            },
        );
    };

    displayMobileFilterDates = () => {
        this.setState(
            {
                showMobileFilter: false,
                showMobileFilterLocations: false,
                showMobileFilterDates: true,
            },
        );
    };

    render() {
        const {
            handleLocalFilterDateOptionSelect,
            selectedDateOption,
            locations,
            handleUserLocationChange,
            selectedUserLocations,
        } = this.props;

        const {
            mobileControlsOpen, showMobileFilter, showMobileFilterLocations, showMobileFilterDates,
        } = this.state;

        return (
            <div className="local-filter">
                <div className="local-filter-container">
                    <LocalDateFilter
                        handleLocalFilterDateOptionSelect={handleLocalFilterDateOptionSelect}
                        selectedDateOption={selectedDateOption}
                    />
                    <LocalUserLocationFilter
                        locations={locations}
                        handleUserLocationChange={handleUserLocationChange}
                        selectedUserLocations={selectedUserLocations}
                    />
                </div>

                <div className="globalfilter-mobile">
                    <Navbar>
                        <NavbarBrand className="globalfilter-mobile-brand">
                            {showMobileFilter ? (
                                <p>Local Filter</p>
                            ) : (
                                <div className="globalfilter-back-button" onClick={this.displayMobileFilter}>
                                    <FontAwesomeIcon icon="arrow-left" className="globalfilter-back-icon" />
                                    <p>Back</p>
                                </div>
                            )}
                        </NavbarBrand>

                        <NavbarToggler onClick={this.toggleMobileControlsAndClose} />

                        <Collapse isOpen={mobileControlsOpen} navbar>
                            <div className="globalfilter-mobile-container">
                                {showMobileFilter
                                    && (
                                        <div>
                                            <div className="globalfilter-locations-container fadein">
                                                <Button
                                                    onClick={this.displayMobileFilterDates}
                                                    className="globalfilter-mobile-button"
                                                >
                                                    {selectedDateOption || "Date Options"}
                                                    <div className="globalfilter-button-caret">
                                                        <FontAwesomeIcon icon="caret-right" />
                                                    </div>
                                                </Button>

                                                <Button
                                                    onClick={this.displayMobileFilterLocations}
                                                    className="globalfilter-mobile-button"
                                                >
                                                    Team Location (
                                                    {selectedUserLocations.length}
                                                    )
                                                    <div className="globalfilter-button-caret">
                                                        <FontAwesomeIcon icon="caret-right" />
                                                    </div>
                                                </Button>
                                            </div>
                                        </div>
                                    )}

                                {showMobileFilterLocations
                                    && (
                                        <ul className="popover show map-controls-popover local-data-review-filter-dropdown  bs-popover-bottom mobile-dropdown">
                                            {locations.map((location) => {
                                                return (
                                                    <li key={location.id}>
                                                        <FormGroup check>
                                                            <Label check className="map-controls-checkbox">
                                                                <Input
                                                                    type="checkbox"
                                                                    onChange={handleUserLocationChange}
                                                                    value={location.id}
                                                                    checked={selectedUserLocations.includes(location.id)}
                                                                />
                                                                {location.name}
                                                            </Label>
                                                        </FormGroup>
                                                    </li>
                                                );
                                            })}
                                        </ul>
                                    )}

                                {showMobileFilterDates
                                    && (
                                        <ul className="map-controls-popover-list fadein">
                                            {DateOptions.ALL_OPTIONS.map((option) => {
                                                return (
                                                    <FormGroup check key={option} inline>
                                                        <Label check>
                                                            <Input
                                                                type="radio"
                                                                name="radio1"
                                                                onChange={handleLocalFilterDateOptionSelect}
                                                                value={option}
                                                                checked={option === selectedDateOption}
                                                            />
                                                            {option}
                                                        </Label>
                                                    </FormGroup>
                                                );
                                            })}
                                        </ul>
                                    )}
                            </div>
                        </Collapse>
                    </Navbar>
                </div>
            </div>
        );
    }
}

export default LocalFilter;
