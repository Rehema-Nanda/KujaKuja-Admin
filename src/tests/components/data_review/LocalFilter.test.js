import React from "react";
import { mount } from "enzyme";

import LocalFilter from "../../../components/data_review/LocalFilter";
import DateOptions from "../../../components/data_review/DateOptions";

describe("LocalFilter component", () => {
    const testProps = {
        handleLocalFilterDateOptionSelect: jest.fn(),
        selectedDateOption: DateOptions.CREATED_AT,
        locations: [{ id: "1", name: "test Location" }],
        handleUserLocationChange: jest.fn(),
        selectedUserLocations: [],
    };

    let wrapper;

    beforeEach(() => {
        wrapper = mount(<LocalFilter {...testProps} />);
    });

    it("renders correctly", () => {
        expect(wrapper.debug()).toMatchSnapshot();
    });

    it("should change the state of mobile controls when toggleMobileControlsAndClose function is called", () => {
        navigator.__defineGetter__("userAgent", () => {
            return "Chrome/75.0.3770.142 Safari/537.3";
        });

        const event = {
            target: {
                style: {},
            },
        };
        const toggle = wrapper.find("NavbarToggler");

        toggle.simulate("click", event);
        expect(wrapper.state().showMobileFilter).toBe(true);
    });

    it("should change showMobileFilterLocations state when displayMobileFilterLocations function is called", () => {
        const button = wrapper.find("Button.globalfilter-mobile-button").at(1);
        button.simulate("click");
        expect(wrapper.state().showMobileFilterLocations).toBe(true);
    });

    it("should change showMobileFilterDates state when displayMobileFilterDates function is called", () => {
        const button = wrapper.find("Button.globalfilter-mobile-button").first();
        button.simulate("click");
        expect(wrapper.state().showMobileFilterDates).toBe(true);
    });

    it("should set showMobileFilter state to true when global filter back button is clicked", () => {
        wrapper.setState({ showMobileFilter: false });
        const back = wrapper.find("div.globalfilter-back-button");
        back.simulate("click");
        expect(wrapper.state().showMobileFilter).toBe(true);
    });
});
