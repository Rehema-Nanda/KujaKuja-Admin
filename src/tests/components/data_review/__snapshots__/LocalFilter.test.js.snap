// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LocalFilter component renders correctly 1`] = `
"<LocalFilter handleLocalFilterDateOptionSelect={[Function: mockConstructor]} selectedDateOption=\\"Created At\\" locations={{...}} handleUserLocationChange={[Function: mockConstructor]} selectedUserLocations={{...}}>
  <div className=\\"local-filter\\">
    <div className=\\"local-filter-container\\">
      <LocalDateFilter handleLocalFilterDateOptionSelect={[Function: mockConstructor]} selectedDateOption=\\"Created At\\">
        <Form className=\\"local-date-container\\" tag=\\"form\\">
          <form className=\\"local-date-container\\">
            <FormGroup tag=\\"div\\">
              <div className=\\"position-relative form-group\\">
                <FormGroup check={true} inline={true} tag=\\"div\\">
                  <div className=\\"position-relative form-check form-check-inline\\">
                    <Label check={true} className=\\"form-label-check\\" tag=\\"label\\" widths={{...}}>
                      <label htmlFor={[undefined]} className=\\"form-label-check form-check-label\\">
                        <Input type=\\"radio\\" name=\\"radio1\\" onChange={[Function: mockConstructor]} value=\\"Created At\\" checked={true}>
                          <input name=\\"radio1\\" onChange={[Function: mockConstructor]} value=\\"Created At\\" checked={true} type=\\"radio\\" className=\\"form-check-input\\" />
                        </Input>
                        Created At
                      </label>
                    </Label>
                  </div>
                </FormGroup>
                <FormGroup check={true} inline={true} tag=\\"div\\">
                  <div className=\\"position-relative form-check form-check-inline\\">
                    <Label check={true} className=\\"form-label-check\\" tag=\\"label\\" widths={{...}}>
                      <label htmlFor={[undefined]} className=\\"form-label-check form-check-label\\">
                        <Input type=\\"radio\\" name=\\"radio1\\" onChange={[Function: mockConstructor]} value=\\"Uploaded At\\" checked={false}>
                          <input name=\\"radio1\\" onChange={[Function: mockConstructor]} value=\\"Uploaded At\\" checked={false} type=\\"radio\\" className=\\"form-check-input\\" />
                        </Input>
                        Uploaded At
                      </label>
                    </Label>
                  </div>
                </FormGroup>
              </div>
            </FormGroup>
          </form>
        </Form>
      </LocalDateFilter>
      <LocalUserLocationFilter locations={{...}} handleUserLocationChange={[Function: mockConstructor]} selectedUserLocations={{...}}>
        <div className=\\"user-location-container\\">
          <div id=\\"UserLocationDropdown\\" className=\\"map-controls-button user-location-button\\" onClick={[Function]}>
            <Badge color=\\"secondary\\" pill={true} tag=\\"span\\">
              <span className=\\"badge badge-secondary badge-pill\\">
                All
              </span>
            </Badge>
             
            Team Location
            <div className=\\"map-controls-button-caret\\">
              <FontAwesomeIcon icon=\\"caret-down\\">
                <i className=\\"fa\\" />
              </FontAwesomeIcon>
            </div>
          </div>
          <mockConstructor placement=\\"bottom\\" isOpen={false} target=\\"UserLocationDropdown\\" toggle={[Function]} className=\\"map-controls-popover\\">
            Popover
          </mockConstructor>
        </div>
      </LocalUserLocationFilter>
    </div>
    <div className=\\"globalfilter-mobile\\">
      <Navbar tag=\\"nav\\" expand={false}>
        <nav className=\\"navbar\\">
          <NavbarBrand className=\\"globalfilter-mobile-brand\\" tag=\\"a\\">
            <a className=\\"globalfilter-mobile-brand navbar-brand\\">
              <p>
                Local Filter
              </p>
            </a>
          </NavbarBrand>
          <NavbarToggler onClick={[Function]} tag=\\"button\\" type=\\"button\\">
            <button onClick={[Function]} type=\\"button\\" className=\\"navbar-toggler\\">
              <span className=\\"navbar-toggler-icon\\" />
            </button>
          </NavbarToggler>
          <Collapse isOpen={false} navbar={true} in={false} mountOnEnter={false} unmountOnExit={false} appear={false} enter={true} exit={true} onEnter={[Function: noop]} onEntering={[Function: noop]} onEntered={[Function: noop]} onExit={[Function: noop]} onExiting={[Function: noop]} onExited={[Function: noop]} tag=\\"div\\" timeout={350}>
            <Transition onExited={[Function: bound onExited]} onExiting={[Function: bound onExiting]} onExit={[Function: bound onExit]} onEntered={[Function: bound onEntered]} onEntering={[Function: bound onEntering]} onEnter={[Function: noop]} timeout={350} exit={true} enter={true} appear={false} unmountOnExit={false} mountOnEnter={false} in={false}>
              <div style={{...}} className=\\"collapse navbar-collapse\\">
                <div className=\\"globalfilter-mobile-container\\">
                  <div>
                    <div className=\\"globalfilter-locations-container fadein\\">
                      <Button onClick={[Function]} className=\\"globalfilter-mobile-button\\" color=\\"secondary\\" tag=\\"button\\">
                        <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"globalfilter-mobile-button btn btn-secondary\\" aria-label={{...}}>
                          Created At
                          <div className=\\"globalfilter-button-caret\\">
                            <FontAwesomeIcon icon=\\"caret-right\\">
                              <i className=\\"fa\\" />
                            </FontAwesomeIcon>
                          </div>
                        </button>
                      </Button>
                      <Button onClick={[Function]} className=\\"globalfilter-mobile-button\\" color=\\"secondary\\" tag=\\"button\\">
                        <button type=\\"button\\" onClick={[Function: bound onClick]} className=\\"globalfilter-mobile-button btn btn-secondary\\" aria-label={{...}}>
                          Team Location (
                          0
                          )
                          <div className=\\"globalfilter-button-caret\\">
                            <FontAwesomeIcon icon=\\"caret-right\\">
                              <i className=\\"fa\\" />
                            </FontAwesomeIcon>
                          </div>
                        </button>
                      </Button>
                    </div>
                  </div>
                </div>
              </div>
            </Transition>
          </Collapse>
        </nav>
      </Navbar>
    </div>
  </div>
</LocalFilter>"
`;
